process/accountProcessor.go

```diff
@@ -87,14 +87,38 @@ func (ap *AccountProcessor) GetAccounts(addresses []string, options common.Accou
 		addressesInShards[shardID] = append(addressesInShards[shardID], address)
 	}
 
+	var wg sync.WaitGroup
+	resultsCh := make(chan map[string]*data.Account, len(addressesInShards))
+	errCh := make(chan error, len(addressesInShards))
+
+	for shID, accounts := range addressesInShards {
+		wg.Add(1)
+		go func(shID uint32, accounts []string) {
+			defer wg.Done()
+			accountsInShard, err := ap.getAccountsInShard(accounts, shID, options)
+			if err != nil {
+				errCh <- err
```

Comment:
possible resource leak & deadlock. 
Example: 2 go routines error. After the first write on the error chan, the call to return on L115 will be called. Then the defer on L107 is called and the L108 will never be completed because the L100 on the second go routine did not finish,
Suggestion to refactor this code